dll/win32/shell32/dialogs/view.cpp

```diff
@@ -860,21 +860,32 @@ ScanAdvancedSettings(SHELLSTATE *pSS, DWORD *pdwMask)
 }
 
 static BOOL CALLBACK
-RefreshBrowsersCallback(HWND hWnd, LPARAM msg)
+PostCabinetMessageCallback(HWND hWnd, LPARAM param)
 {
+    MSG &data = *(MSG*)param;
     WCHAR ClassName[100];
     if (GetClassNameW(hWnd, ClassName, _countof(ClassName)))
     {
         if (!wcscmp(ClassName, L"Progman") ||
             !wcscmp(ClassName, L"CabinetWClass") ||
             !wcscmp(ClassName, L"ExploreWClass"))
         {
-            PostMessage(hWnd, WM_COMMAND, FCIDM_DESKBROWSER_REFRESH, 0);
+            PostMessage(hWnd, data.message, data.wParam, data.lParam);
         }
     }
     return TRUE;
 }
 
+void
+PostCabinetMessage(UINT Msg, WPARAM wParam, LPARAM lParam)
+{
+    MSG data;
+    data.message = Msg;
+    data.wParam = wParam;
+    data.lParam = lParam;
+    EnumWindows(PostCabinetMessageCallback, (LPARAM)&data);
```

Comment:
Note to self: as long as the data values (stored in w/lParam) are "scalar" values, the resulting PostMessage call is OK. But if they are pointers to data whose lifetime isn't correctly controlled (and freed before the posted message is fully handled), problems can arise.

Reply:
Yes but it has to be that way. SendMessage could lock us up. SendMessageTimeout can also break pointers.

None of that matters because we are supposed to support multiple Explorer.exe processes so any message parameter would have to be a SHAllocShared if you need to pass a string/PIDL.

Reply:
OK, as long as the problem is known :+1: